src/patterns/ec2-app/base.ts

```diff
@@ -159,14 +136,7 @@ function restrictedCidrRanges(ranges: IPeer[]) {
  *   access: { scope: AccessScope.PUBLIC },
  *   instanceType: InstanceType.of(InstanceClass.T4G, InstanceSize.MEDIUM),
  *   certificateProps:{
- *     [Stage.CODE]: {
- *       domainName: "code-guardian.com",
- *       hostedZoneId: "id123",
- *     },
- *     [Stage.PROD]: {
- *       domainName: "prod-guardian.com",
- *       hostedZoneId: "id124",
- *     },
+ *     domainName: "prod-guardian.com",
```

Comment:
As this section of the Typedocs will be the entrypoint for many users, I wonder if we should demonstrate how to instantiate this pattern in a stage-aware fashion. I think the usage example from the release notes (and PR description) is extremely helpful - so it would be great to find a way to surface it for all users, rather than just users who are upgrading.

Although it's not technically part of the pattern anymore, it's still probably the most typical use-case and it would be good to give some guidance on the type of properties that we expect users to vary based on stage (since this is no longer encoded).

Alternatively we could keep it simple here and hardcode a string, but link to a more general 'Here's how to set up a stage aware pattern' that we can point to from all of the different patterns?

Reply:
Added.